osu.Game/Graphics/UserInterface/OsuTabControl.cs

```diff
@@ -98,6 +98,14 @@ protected override void UpdateAfterChildren()
                 strip.Width = Interpolation.ValueAt(Math.Clamp(Clock.ElapsedFrameTime, 0, 1000), strip.Width, StripWidth, 0, 500, Easing.OutQuint);
         }
 
+        public void SelectTabWithSound(T item)
+        {
+            Current.Value = item;
+
+            if (SelectedTab is OsuTabItem selectedTab)
+                selectedTab.TabSelectSounds.PlayClickSample();
```

Comment:
I dunno. The entire point of `HoverClickSounds` is to handle sample playback automatically. At the point this has to be done for correct UX in selected usages, is there even any point left to it?

This sort of thing is breaching multiple levels of abstraction and patching a system that - given the requirements as I understand them - is fundamentally broken.

At that point I'd say that the possible alternative of deleting `HoverClickSounds` and replacing with standard inline sample playbacks should be looked at. Or playing a sample on keyboard interaction in `Editor` even. Because this is just too weird to live IMO.

And yes I know the usage count means that getting rid of this is not going to be pretty. But I'm not looking forward to having to pepper `PlayClickSample()` calls everywhere or wondering whether a particular method of selecting a tab is going to play a sound or not.

Reply:
Can we just do this in line with how we did it elsewhere and expose framework support?

ie.

https://github.com/ppy/osu/blob/e081fa48a21d2b89ab9d956e62495b95f4ea9c1f/osu.Game/Graphics/UserInterface/OsuSliderBar.cs#L51-L58

Reply:
Yeah, that's what I meant in https://github.com/ppy/osu/pull/27448#issuecomment-1975380357.

Reply:
So this is waiting on changes from you then..?